pandas_gbq/tests/test_gbq.py

```diff
@@ -1154,6 +1156,45 @@ def test_google_upload_errors_should_raise_exception(self):
             gbq.to_gbq(bad_df, self.destination_table + test_id,
                        _get_project_id(), private_key=_get_private_key_path())
 
+    def test_upload_chinese_unicode_data(self):
+        test_id = "2"
+        test_size = 6
+        df = DataFrame(np.random.randn(6, 4), index=range(6),
+                       columns=list('ABCD'))
+        df.A = u'信用卡'
+
+        gbq.to_gbq(df, self.destination_table + test_id, _get_project_id(),
+                   chunksize=10000)
+
+        result = gbq.read_gbq("SELECT COUNT(*) AS num_rows FROM {0}".format(
+            self.destination_table + test_id),
+            project_id=_get_project_id())
+
+        assert result['num_rows'][0] == test_size
+        tm.assert_series_equal(result['A'], df['A'])
+
+    def test_upload_other_unicode_data(self):
+        test_id = "3"
+        test_size = 3
+        df = DataFrame({
+            'string': ['Skywalker™', 'lego', 'hülle'],
+            'integer': [200, 300, 400],
+            'Date': [
+                '2017-12-13 17:40:39', '2017-12-13 17:40:39',
+                '2017-12-13 17:40:39'
+            ]
+        })
+
+        gbq.to_gbq(df, self.destination_table + test_id, _get_project_id(),
+                   chunksize=10000)
+
+        result = gbq.read_gbq("SELECT COUNT(*) AS num_rows FROM {0}".format(
+            self.destination_table + test_id),
+            project_id=_get_project_id())
+
+        assert result['num_rows'][0] == test_size
+        tm.assert_series_equal(result['string'], df['string'])
```

Comment:
You'll want to either sort the `result` dataframe by integer or do an `order by` in the query to ensure that the assertion will pass (otherwise, the rows of result could be in a different order than your original `df`).